vendor/bat-native-ads/src/bat/ads/internal/purchase_intent/funnel_sites.cc

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/purchase_intent/funnel_sites.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+
+#include "url/gurl.h"
+
+namespace ads {
+
+FunnelSites::FunnelSites() = default;
+FunnelSites::~FunnelSites() = default;
+
+FunnelSiteInfo FunnelSites::MatchFunnelSite(
+    const std::string& url) {
+  auto visited_url = GURL(url);
```

Comment:
nit: `const std::string` instead of `auto`

Reply:
I think the type should be `GURL`?

Reply:
ah yes sorry `GURL`